pkg/apis/autoscaling/v1/fleetautoscaler_test.go

```diff
@@ -456,6 +457,160 @@ func TestFleetAutoscalerListValidateUpdate(t *testing.T) {
 	}
 }
 
+func TestFleetAutoscalerScheduleValidateUpdate(t *testing.T) {
+	t.Parallel()
+
+	modifiedFAS := func(f func(*FleetAutoscalerPolicy)) *FleetAutoscaler {
+		fas := scheduleFixture()
+		f(&fas.Spec.Policy)
+		return fas
+	}
+
+	testCases := map[string]struct {
+		fas          *FleetAutoscaler
+		featureFlags string
+		wantLength   int
+		wantField    string
+	}{
+		"end time before current time": {
+			fas: modifiedFAS(func(fap *FleetAutoscalerPolicy) {
+				fap.Schedule.Between.End = metav1.Date(2024, 0o7, 0o3, 15, 59, 59, 0, time.UTC)
```

Comment:
As a suggestion to make the tests more readable, just add a mustParseDate function that takes an RFC3339 field.

Reply:
Sg, for some odd reason k8's time representation enforces octal literal style.